src/main/java/org/openrewrite/java/migrate/guava/NoGuavaPrimitiveAsList.java

```diff
@@ -0,0 +1,88 @@
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.guava;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesMethod;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.template.internal.AbstractRefasterJavaVisitor;
+import org.openrewrite.java.tree.J;
+
+public class NoGuavaPrimitiveAsList extends Recipe {
+
+    static final MethodMatcher METHOD_MATCHER = new MethodMatcher("com.google.common.primitives.* asList(..)");
+
+    @Override
+    public String getDisplayName() {
+        return "`com.google.common.primitives asList` to `Arrays.asList`";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Migrate from Guava `com.google.common.primitives asList` to `Arrays.asList`.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        JavaVisitor<ExecutionContext> javaVisitor = new AbstractRefasterJavaVisitor() {
+
+            @Override
+            public J visitMethodInvocation(J.MethodInvocation elem, ExecutionContext ctx) {
+                if (METHOD_MATCHER.matches(elem)) {
+                    maybeRemoveImport("com.google.common.primitives.*");
+                    maybeRemoveImport("com.google.common.primitives.Booleans");
+                    maybeRemoveImport("com.google.common.primitives.Chars");
+                    maybeRemoveImport("com.google.common.primitives.Doubles");
+                    maybeRemoveImport("com.google.common.primitives.Floats");
+                    maybeRemoveImport("com.google.common.primitives.Longs");
+                    maybeRemoveImport("com.google.common.primitives.Ints");
+                    maybeRemoveImport("com.google.common.primitives.Shorts");
+                    maybeRemoveImport("com.google.common.primitives.Bytes");
+                    maybeAddImport("java.util.Arrays");
+                    StringBuilder sb = new StringBuilder();
+                    sb.append("Arrays.asList(");
+                    for (int i = 0; i < elem.getArguments().size(); i++) {
+                        sb.append("#{any()}");
+                        if (i != elem.getArguments().size() - 1) {
+                            sb.append(", ");
+                        }
+                    }
```

Comment:
This alternative is a little more expressive perhaps.
```suggestion
                    sb.append(String.join(",", Collections.nCopies(elem.getArguments().size(), "#{any()}")));
```

Reply:
That might then also remove the need for the StringBuilder, as that part can be inlined in `"Arrays.asList(" + joined + ')'`.